include/dr/mhp/algorithms/inclusive_exclusive_scan_impl.hpp

```diff
@@ -110,24 +139,27 @@ auto inclusive_scan_impl_(R &&r, O &&d_first, BinaryOp &&binary_op,
         auto local_in = dr::ranges::__detail::local(global_in);
         auto local_out = rng::views::take(
             dr::ranges::__detail::local(global_out), rng::size(local_in));
-        // dr::drlog.debug("rebase before: {}\n", local_out);
+        auto local_out_begin_direct =
+            detail::direct_iterator(local_out.begin());
+        *local_out_begin_direct = offset;
```

Comment:
I wonder if this line can work. We are in host code (not a kernel), and assignment is done on data allocated on GPU. Maybe it will work with shared_allocator allocated memory, but we are switching to device_allocated memory and then it may be broken.

Reply:
I've asked in other comment to leave FIXME about this.